### PR TITLE
charmstore: add NoIndexes config option

### PR DESCRIPTION
**Pull request by rogpeppe**
As an example, using this flag reduces the time taken to test `github.com/juju/charmstore-client/cmd/charm/charmcmd` from 3m23s to 1m3s.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2251/

**Review by fabricematrat — APPROVED** [review]

File: internal/charmstore/server.go [hunk 1]
```diff
@@ -125,6 +125,11 @@ type ServerParams struct {
 	// the blobstore garbage collector worker.
 	RunBlobStoreGC bool
 
+	// NoIndexes specifies that none of the MongoDB indexes should be
+	// created. This speeds up initialization (useful for tests) but should
+	// never be  in production.
```
Comment (mhilton) [inline internal/charmstore/server.go]: There is a spare ' ' in here.

**Review by mhilton — APPROVED** [review]
LGTM

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2252/

**Pull request merged**